src/rust/wpcomics/sources/readcomicsfree/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+#![no_std]
+pub mod helper;
+use crate::helper::*;
+use aidoku::{
+	error::Result, prelude::*, std::net::Request, std::String, std::Vec, Chapter, DeepLink, Filter,
+	FilterType, Listing, Manga, MangaPageResult, MangaViewer, Page,
+};
+use wpcomics_template::{
+	helper::{get_tag_id, urlencode},
+	template::{self, WPComicsSource},
+};
+
+fn get_instance() -> WPComicsSource {
+	WPComicsSource {
+		base_url: "https://readcomicsfree.com",
+		listing_mapping: listing_map,
+
+		manga_cell_image: "",
+
+		manga_details_status_transformer: |str| String::from(str.trim()),
+
+		chapter_skip_first: true,
+		chapter_date_selector: "div.col-xs-3",
+
+		manga_viewer_page_url_suffix: "/all",
+		..Default::default()
+	}
+}
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut title: String = String::new();
+	let mut genre: String = String::new();
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				title = urlencode(filter.value.as_string()?.read());
+			}
+			_ => match filter.name.as_str() {
+				"Genre" => {
+					genre = get_tag_id(filter.value.as_int().unwrap_or(0));
+				}
+				_ => continue,
+			},
+		}
+	}
+	get_instance().get_manga_list(get_search_url(
+		String::from("https://readcomicsfree.com"),
+		title,
+		genre,
+		page,
+	))
+}
+
+#[get_manga_listing]
+fn get_manga_listing(listing: Listing, page: i32) -> Result<MangaPageResult> {
+	get_instance().get_manga_listing(listing, page)
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+	get_instance().get_manga_details(id, MangaViewer::Ltr)
+}
+
+#[get_chapter_list]
+fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
+	get_instance().get_chapter_list(id)
+}
+
+#[get_page_list]
+fn get_page_list(id: String) -> Result<Vec<Page>> {
+	get_instance().get_page_list(id)
+}
+
+#[modify_image_request]
+fn modify_image_request(request: Request) {
+	template::modify_image_request(
+		String::from("https://readcomicsfree.com"),
+		String::from("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36 Edg/101.0.1210.39"),
+		request,
+	)
```

Comment:
Since the site uses CloudFlare for image requests too, you have to either use the same user agent as the other requests, or let Aidoku use the default user agent. Since the images don't seem to need the Referer header, you might as well not do anything here.
```suggestion
```